x-pack/plugins/apm/server/routes/agent_explorer/get_agents_filters.ts

```diff
@@ -0,0 +1,96 @@
+import { AGENT_NAME, AGENT_VERSION, SERVICE_LANGUAGE_NAME, SERVICE_NAME, SERVICE_NODE_NAME } from "@kbn/apm-plugin/common/elasticsearch_fieldnames";
+import { environmentQuery } from "@kbn/apm-plugin/common/utils/environment_query";
+import { ProcessorEvent } from "@kbn/observability-plugin/common";
+import { rangeQuery } from "@kbn/observability-plugin/server";
+import { RandomSampler } from "../../lib/helpers/get_random_sampler";
+import { Setup } from "../../lib/helpers/setup_request";
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+export async function getAgentsFilters({
```

Comment:
You could check [SuggestionsSelect](https://github.com/elastic/kibana/blob/b31c1d6462fe1b893f0791100e257dc85121a8cf/x-pack/plugins/apm/public/components/shared/suggestions_select/index.tsx#L28)